File: Cplusplus/소수찾기.cpp
//22.01.18 소수찾기 by CJHofficial

#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(int n) {
    int answer = 0;
    int prime[n + 1];
    for (int i = 2; i <= n; i++) prime[i] = 1;
    for (int i = 2; i <= n; i++) {
        if (prime[i] == 0) continue;
        for (int j = i * 2; j <= n; j += i) {
            prime[j] = 0;
        }
    }
    for (int i = 2; i <= n; i++) {
        if (prime[i] != 0) answer += 1;
    }
    return answer;
}

File: Python/level2/소수 찾기.py
import itertools

def solution(numbers):
    answer = set()
    for index in range(1, len(numbers) + 1):
        for permutation in itertools.permutations(list(numbers), index):
            number = int(''.join(permutation))
            for div in range(2,int(number ** (1/2)) + 1):
                if number % div == 0:
                    break
            else:
                if number != 1 and number != 0:
                    answer.add(number)
    return len(answer)
